internal/services/compute/orchestrated_virtual_machine_scale_set.go

```diff
@@ -496,22 +497,65 @@ func OrchestratedVirtualMachineScaleSetDataDiskSchema() *pluginsdk.Schema {
 					Default:  false,
 				},
 
-				// TODO 3.0 - change this to ultra_ssd_disk_iops_read_write
+				//TODO: Remove the attribute below in 4.0
 				"disk_iops_read_write": {
 					Type:     pluginsdk.TypeInt,
 					Optional: true,
 					Computed: true,
+					ConflictsWith: func() []string {
+						if features.ThreePointOhBeta() {
+							return []string{"ultra_ssd_disk_iops_read_write"}
+						}
+						return []string{}
+					}(),
+					Deprecated: func() string {
+						if features.ThreePointOhBeta() {
+							return "This property has been renamed to `ultra_ssd_disk_iops_read_write` and will be removed in v3.0 of the provider"
+						}
+						return ""
+					}(),
 				},
 
-				// TODO 3.0 - change this to ultra_ssd_disk_iops_read_write
+				//TODO: Remove the attribute below in 4.0
 				"disk_mbps_read_write": {
 					Type:     pluginsdk.TypeInt,
 					Optional: true,
 					Computed: true,
+					ConflictsWith: func() []string {
+						if features.ThreePointOhBeta() {
+							return []string{"ultra_ssd_mbps_read_write"}
+						}
+						return []string{}
+					}(),
+					Deprecated: func() string {
+						if features.ThreePointOhBeta() {
+							return "This property has been renamed to `ultra_ssd_disk_mbps_read_write` and will be removed in v3.0 of the provider"
+						}
+						return ""
+					}(),
 				},
 			},
 		},
 	}
+
+	o := out.Elem.(*pluginsdk.Resource).Schema
+	if features.ThreePointOhBeta() {
+		o["ultra_ssd_disk_iops_read_write"] = &pluginsdk.Schema{
+			Type:          pluginsdk.TypeInt,
+			Optional:      true,
+			Computed:      true,
+			ConflictsWith: []string{"disk_iops_read_write"},
+		}
+
+		o["ultra_ssd_mbps_read_write"] = &pluginsdk.Schema{
+			Type:          pluginsdk.TypeInt,
+			Optional:      true,
+			Computed:      true,
+			ConflictsWith: []string{"disk_mbps_read_write"},
+		}
+	}
+
+	return out
```

Comment:
is there a reason we are adding these in 3,0 and not just adding the new properties  deprecating the old ones and removing in 3.0/waiting fort 4.0 for the removable?

Reply:
Alright, based on this and https://github.com/hashicorp/terraform-provider-azurerm/pull/15390#discussion_r808695246 I came up with the approach you can see in the commint below 

(https://github.com/hashicorp/terraform-provider-azurerm/pull/15340/commits/00c3174648a958afe55e7f3ad9f711b67896ae8e)

tl;dr
2.x: There is a deprecation notice to the existing fields
3.x: The fields are renamed.